lib/sails-migrations/helpers/config_loader.js

```diff
@@ -51,6 +45,20 @@ function getConfigFromSailsConfig(sailsConfig) {
   return result;
 }
 
+function getConnectionFromSailsConfig(sailsConfig) {
+  if (typeof sailsConfig.defaultAdapter.config.url !== 'undefined' && sailsConfig.defaultAdapter.config.url !== null) {
+    return sailsConfig.defaultAdapter.config.url;
```

Comment:
In knex, you have two options to create the db connection. One option is to pass the parameters in a object and the other one is to specify the connection url (http://knexjs.org/#Installation-client). If specified, the connection url should contains all the required information to connect to the database. In my sails app, I use the db connection url as given by heroku.